### PR TITLE
fix(ktable): more row click fix [beta]

### PR DESCRIPTION
**Pull request by kaiarrowood**
### Summary
Fix an issue where clicking a popover triggered from a button containing only an icon throwing an error on click.

#### Changes made:
- Fix handling of `className` when an object
- Extend row click logic to query parents
- Docs example update

### Vue 3 Upgrade

**If your PR contains code that needs to be ported to the `beta` branch, please add the [`port to beta branch`](https://github.com/Kong/kongponents/labels/port%20to%20beta%20branch) label to your PR.**

If you have questions, tag `@adamdehaven` or `@kaiarrowood`.

### PR Checklist
- [x] Does not introduce dependencies
- [x] **Functional:** all changes do not break existing APIs and if so, bump major version.
- [x] **Tests pass:** check the output of yarn test packages/<Kongponent>
- [x] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
- [x] **Framework style:** abides by the essential rules in Vue's style guide
- [x] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs), or leftover comments
- [x] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

File: src/components/KTable/KTable.vue [hunk 1]
```diff
@@ -554,9 +554,32 @@ export default defineComponent({
...
+            if (typeof targetClasses === 'string' || Array.isArray(targetClasses)) {
+              isPopoverContent = targetClasses.includes('k-popover')
+            } else if (typeof targetClasses === 'object') {
+              isPopoverContent = Object.keys(targetClasses).includes('k-popover')
+            }
+
+            // check parent for popover class
+            if (e.target.closest('.k-popover-content') !== null) {
+              isPopoverContent = true
+            }
```
Comment (adamdehaven) [inline src/components/KTable/KTable.vue:571]: Just checking, should this always eval to true if the previous if block is still false?

File: src/components/KTable/KTable.vue [hunk 2]
```diff
@@ -554,9 +554,32 @@ export default defineComponent({
...
+
+            // check parent of target is not an ignored elem
+            for (let i = 0; i < ignoredElements.length; i++) {
+              if (e.target.closest(ignoredElements[i]) !== null) {
+                isIgnored = true
+                break
+              }
+            }
```
Comment (adamdehaven) [inline src/components/KTable/KTable.vue:579]: This says you're checking the element's parent; however, if you're using `closest` this is traversing the element and ALL of its parents, all the way up the DOM. Is this intended?

Comment (kaiarrowood) [reply src/components/KTable/KTable.vue:579]: Yes

File: src/components/KTable/KTable.vue (hunk 1 above)
Comment (kaiarrowood) [reply src/components/KTable/KTable.vue:571]: Yes

**Review by adamdehaven — APPROVED** [review]
lgtm

**Pull request merged**